src/ethereum/EthereumEngine.ts

```diff
@@ -739,13 +744,34 @@ export class EthereumEngine extends CurrencyEngine<
   // Public methods
   // ****************************************************************************
 
-  async startEngine(): Promise<void> {
-    this.currencyEngineCallbacks.onSubscribeAddresses([
-      {
-        address: this.walletLocalData.publicKey,
-        checkpoint: this.walletLocalData.highestTxBlockHeight.toString()
-      }
-    ])
+  async startEngine(opts: EdgeStartEngineOptions = {}): Promise<void> {
+    const realAddress = this.walletLocalData.publicKey
+
+    // Sync in-memory decoy addresses with what core has saved
+    if (opts.subscribedAddresses != null) {
+      this.subscribedAddresses = opts.subscribedAddresses
+    }
+
+    // Ensure that the real address is in the list
+    this.subscribedAddresses.unshift({
+      address: realAddress,
+      checkpoint: this.walletLocalData.highestTxBlockHeight.toString()
+    })
+
+    // Efficient deduplication of subscribed addresses
+    const seenAddresses = new Set<string>()
+    this.subscribedAddresses = this.subscribedAddresses.filter(a => {
```

Comment:
Why would you have duplicates? The code that adds addresses already ensures you don't have dupes.

Reply:
This is to deduplicate the real address, but if we're not going to add it, then this can be removed.